zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLFuncVisitor.scala

```diff
@@ -179,9 +181,19 @@ object AQLFuncVisitor {
             case Some(name) => (hierarchyField.innerType, s"$aqlExpr.${name._2}")
             case None => throw new QueryParsingException(s"could not find the level tag ${level.levelTag} in hierarchy field ${field.name}")
           }
-        case GeoCellTenth => ???
-        case GeoCellHundredth => ???
-        case GeoCellThousandth => ???
+        case GeoCellTenth =>
+          if (field.dataType != DataType.Point) throw new QueryParsingException("Geo-cell requires a point")
+          val origin = s"create-point(0.0,0.0)"
+          (DataType.Point, s"get-points(spatial-cell(${aqlExpr}, $origin, 0.1, 0.1))[0]")
+        case GeoCellHundredth =>
+          if (field.dataType != DataType.Point) throw new QueryParsingException("Geo-cell requires a point")
+          val origin = s"create-point(0.0,0.0)"
+          (DataType.Point, s"get-points(spatial-cell(${aqlExpr}, $origin, 0.01, 0.01))[0]")
+        case GeoCellThousandth =>
+          if (field.dataType != DataType.Point) throw new QueryParsingException("Geo-cell requires a point")
```

Comment:
can you create a small function and just pass the `0.1, 0.01, 0.001` into it?  In which case we can save many lines.